src/content/docs/docs/integrations/google-genai.mdx

```diff
@@ -952,96 +952,610 @@ TTS models automatically detect the input language. Supported languages include
+- `gemini-3-flash-preview` - Fast and intelligent model for high-volume tasks
+- `gemini-3-pro-image-preview` - Supports image generation outputs
+
+**Gemini 2.5 Series** - Latest stable models with advanced reasoning and multimodal capabilities:
+- `gemini-2.5-pro` - Most capable stable model for complex tasks
+- `gemini-2.5-flash` - Fast and efficient for most use cases
+- `gemini-2.5-flash-lite` - Lightweight version for simple tasks
+- `gemini-2.5-flash-image` - Supports image generation outputs
+
+**Gemma 3 Series** - Open models for various use cases:
+- `gemma-3-27b-it` - Large instruction-tuned model
+- `gemma-3-12b-it` - Medium instruction-tuned model
+- `gemma-3-4b-it` - Small instruction-tuned model
+- `gemma-3-1b-it` - Tiny instruction-tuned model
+- `gemma-3n-e4b-it` - Efficient 4-bit model
+
+:::note
+See the [Google Generative AI models documentation](https://ai.google.dev/gemini-api/docs/models) for a complete list of available models and their capabilities.
+:::
+
+### Basic Usage
+
+```go
+resp, err := genkit.Generate(ctx, g,
+	ai.WithModelName("googleai/gemini-2.5-flash"),
+	ai.WithPrompt("Explain how neural networks learn in simple terms."),
+)
 ```
 
+### Model References and Configuration
+
+You can reference models in two ways:
+
+- **`ai.WithModelName(name)`**: Resolves a model by its strong identifier (e.g., `"googleai/gemini-2.5-flash"`). This is the recommended way for basic calls.
+- **`googlegenai.GoogleAIModelRef(name, config)`**: Creates a static reference that includes a specific configuration.
+
+You can provide configuration either when referencing the model or per-request to the `genkit.Generate` call:
+
 ```go
-g := genkit.Init(context.Background(), genkit.WithPlugins(&googlegenai.GoogleAI{}))
+import "google.golang.org/genai"
+
+config := &genai.GenerateContentConfig{
+	Temperature: genai.Ptr[float32](0.5),
+}
+
+// Option 1: Use a model reference with "baked-in" config
+modelRef := googlegenai.GoogleAIModelRef("gemini-2.5-flash", config)
+resp, err := genkit.Generate(ctx, g, ai.WithModel(modelRef), ai.WithPrompt("..."))
+
+// Option 2: Pass configuration per-request
+resp, err = genkit.Generate(ctx, g,
+	ai.WithModelName("googleai/gemini-2.5-flash"),
+	ai.WithConfig(config), // Pass config explicitly
+	ai.WithPrompt("..."),
+)
 ```
 
-The plugin requires an API key for the Gemini API, which you can get from
-[Google AI Studio](https://aistudio.google.com/app/apikey).
+### Structured Output
+
+Gemini models support structured output generation, which guarantees that the model output will conform to a specified JSON schema.
+
+```go
+type CharacterProfile struct {
+	Name string `json:"name"`
+	Bio  string `json:"bio"`
+	Age  int    `json:"age"`
+}
+
+var profile CharacterProfile
+resp, err := genkit.Generate(ctx, g,
+	ai.WithModel(model),
+	ai.WithPrompt("Generate a profile for a fictional character"),
+	ai.WithOutputSchema(profile),
```

Comment:
This is not valid. Either `WithOutputType(CharacterProfile{})` or `char, resp, err := genkit.GenerateData[CharacterProfile](...)`.